### PR TITLE
Commitlint Config

### PR DESCRIPTION
**Pull request by oliviaoddo**

### CODE REVIEW

File: src/lint.js [hunk 1]
```diff
@@ -1,16 +1,46 @@
...
+    fs.unlinkSync('src/commitlint-custom.config.js')
+  } catch(e){
+    context.log('Existing custom config not present')
+  }
+
+  // check if pr contains custom config file and write it to src
+  try {
+    const customConfig = await context.github.repos.getContent(context.repo(params))
+    let buff = Buffer.from(customConfig.data.content, 'base64')
+    fs.writeFileSync('src/commitlint-custom.config.js', buff)
```
Comment (gdi2290) [inline src/lint.js]: why are we writing a file?

File: src/get-config.js [hunk 2]
```diff
@@ -8,21 +10,22 @@ async function getConfig(context){
     ref: pr.head.ref
   }
 
+  let decodedCustomConfig;
   // check if pr contains custom config file
   try {
     const customConfig = await context.github.repos.getContent(context.repo(params))
-    const decoded = Base64.decode(customConfig.data.content);
+    decodedCustomConfig = eval(atob(customConfig.data.content))
```
Comment (gdi2290) [inline src/get-config.js]: don't use eval

File: src/get-config.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    ref: pr.head.ref
+  }
+
+  let decodedCustomConfig;
+  // check if pr contains custom config file
+  try {
+    const customConfig = await context.github.repos.getContent(context.repo(params))
+    decodedCustomConfig = JSON.parse(atob(customConfig.data.content))
+  } catch (e) {
+    context.log('Using default config')
```
Comment (gdi2290) [inline src/get-config.js]: remove log

File: src/get-config.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    return defaultConfig;
+  }
+
+  if(decodedCustomConfig.extends){
+    context.log('Extends not supported, using default config.')
```
Comment (gdi2290) [inline src/get-config.js]: remove log put these messages in the details page

File: src/repoCreated.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+function repoCreated() {
```
Comment (gdi2290) [inline src/repoCreated.js]: whats this for?

Comment (oliviaoddo) [reply src/repoCreated.js]: lol accidentally checked that into git

**Pull request merged**